### ISSUE TITLE
Tab completion crashing IRkernel

### ISSUE THREAD

**Issue by pgcudahy** [vatlab/sos#1506]
Hello Bo, I'm having an issue with tab completion in SoS notebooks. In a standard R notebook (with IRkernel) it works fine, but if I hit tab in an R cell in SoS then the busy indicator lights up and the whole notebook hangs. Sometimes an interrupt will fix it, but often I have to restart the kernel. The jupyter logs aren't too helpful, even with the `--debug` flag. They show

```
Nov 22 07:42:46 WhitePlague jupyter[1732]: [D 07:42:46.487 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: status (busy)
Nov 22 07:42:46 WhitePlague jupyter[1732]: [D 07:42:46.488 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: status (idle)
Nov 22 07:42:47 WhitePlague jupyter[1732]: [D 07:42:47.236 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: status (busy)
Nov 22 07:42:47 WhitePlague jupyter[1732]: [D 07:42:47.238 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: status (idle)
Nov 22 07:42:49 WhitePlague jupyter[1732]: [D 07:42:49.477 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: status (busy)
Nov 22 07:42:50 WhitePlague jupyter[1732]: [D 07:42:50.455 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: comm_msg
Nov 22 07:42:50 WhitePlague jupyter[1732]: [D 07:42:50.720 NotebookApp] activity on 05eddf7d-25f3-4ee5-8955-29b49e50becb: stream
Nov 22 07:42:51 WhitePlague jupyter[1732]: [D 07:42:51.217 NotebookApp] 304 GET /notebook/static/base/images/favicon-busy-1.ico (127.0.0.1) 1.410000ms
```
It also hangs in bash cells but tab completion works in python cells. Any ideas on how to further debug this?

**Comment by BoPeng** [comment]
@pgcudahy Sorry that I somehow missed this ticket. A lot has changed since your report and I just released sos-notebook 0.24.0, along with new versions of all language modules. Could you please check if this problem still exists after you upgrade to the newest release? I just tried to complete on a R kernel in jupyter lab and everything seems to be working.


<img width="2305" alt="image" src="https://user-images.githubusercontent.com/9889312/219900818-13d0fd20-e1ab-4626-973d-b222c7231a17.png">

**Comment by pgcudahy** [comment]
Works well now! Thanks, and I'll go ahead and close the issue.
